tests/suites/test_suite_pkwrite.function

```diff
@@ -72,7 +73,8 @@ static void pk_write_check_common(char *key_file, int is_public_key, int is_der)
     unsigned char *buf = NULL;
     unsigned char *check_buf = NULL;
     unsigned char *start_buf;
-    size_t buf_len, check_buf_len;
+    size_t buf_len, check_buf_len, wrong_buf_len = 1;
```

Comment:
I'd like to test with a buffer that's just one byte too short, in case we've messed up a size calculation somewhere. In fact, to be more comprehensive, we could test with all possible shorter sizes (IIRC we do that in a few ASN.1 write tests).